Sources/NIOSSL/SSLPKCS12Bundle.swift

```diff
@@ -158,6 +158,86 @@ public struct NIOSSLPKCS12Bundle: Hashable {
 
 extension NIOSSLPKCS12Bundle: Sendable {}
 
+extension NIOSSLPKCS12Bundle {
+    /// Create a PKCS#12 file containing the given certificates.
+    ///
+    /// The first certificate of the `certificates` array will be considered the "primary" certificate for
+    /// this PKCS#12, and `privateKey` must be its corresponding private key.
+    /// The other certificates included in `certificates`, if any, will be considered as additional
+    /// certificates in the certificate chain.
+    ///
+    /// - Parameters:
+    ///   - certificates: An array of certificates to include in this PKCS#12. Must contain at least one certificate.
+    ///   - privateKey: The private key associated to the first certificate in `certificates`.
+    ///   - passphrase: The password with which to protect this PKCS#12 file.
+    ///   - name: The name to give this PKCS#12 file.
+    /// - Returns: An array of bytes making up the PKCS#12 file.
+    public static func makePKCS12<Bytes: Collection>(
+        certificates: [NIOSSLCertificate],
+        privateKey: NIOSSLPrivateKey,
+        passphrase: Bytes,
+        name: Bytes
+    ) throws -> [UInt8] where Bytes.Element == UInt8 {
+        guard let mainCertificate = certificates.first else {
+            preconditionFailure("At least one certificate must be provided")
+        }
+
+        let certificateChainStack = CNIOBoringSSL_sk_X509_new(nil)
+        for additionalCertificate in certificates.dropFirst() {
+            let result = additionalCertificate.withUnsafeMutableX509Pointer { certificate in
+                CNIOBoringSSL_sk_X509_push(certificateChainStack, certificate)
```

Comment:
This gets the refcount wrong. We need to increase the refcount of the certificate pointer here, using `X509_up_ref`

Reply:
Ah, the joys of manual memory management. 
Should we decrease the refcount after we've created the PKCS12?

Reply:
Yes, we're also missing the free of the `sk_509`. Done correctly, the free call is `sk_X509_pop_free` with the second argument as `X509_free`.